src/kanata/linux.rs

```diff
@@ -31,15 +31,20 @@ impl Kanata {
         Kanata::set_repeat_rate(&k.defcfg_items)?;
         drop(k);
 
+        let scroll_wheel_mapped = MAPPED_KEYS.lock().contains(&OsCode::MouseWheelUp)
+            || MAPPED_KEYS.lock().contains(&OsCode::MouseWheelDown)
+            || MAPPED_KEYS.lock().contains(&OsCode::MouseWheelLeft)
+            || MAPPED_KEYS.lock().contains(&OsCode::MouseWheelRight);
+
         loop {
             let events = kbd_in.read().map_err(|e| anyhow!("failed read: {}", e))?;
             log::trace!("{events:?}");
 
             for in_event in events.into_iter() {
-                let key_event = match KeyEvent::try_from(in_event) {
+                let supported_in_event = match SupportedInputEvent::try_from(in_event) {
```

Comment:
It seems inefficient and like a lot of code to have to convert here:

```
SupportedInputEvent::try_from(in_event)
```

and then convert again below:

```
tx.try_send(supported_in_event.try_into()
```

Maybe this could handle key events the same as before and special-case the mouse handling? The code is already doing that below anyway, in the long body of `match supported_in_event {`, so unifying the two event types doesn't seem useful.

Reply:
Sure, I can optimize it. But I need to add additional items for hi-res scrolls to OsCode only for this code to work:

https://github.com/jtroo/kanata/blob/7e646fb853cb4934627cda731a4c71791afb81d6/src/kanata/linux.rs#L117-L133

Reply:
One question though: if `SupportedInputEvent` were to be removed, how are we supposed to get the scroll distance here?

https://github.com/jtroo/kanata/blob/37f16129572fce319bc9c700de2f2c45f7bc359b/src/kanata/linux.rs#L96

Reply:
done